internal/config/config.go

```diff
@@ -73,8 +75,19 @@ type Config struct {
 	LogLevel slog.Level `yaml:"log"`
 }
 
+var reEnv = regexp.MustCompile(`\$\{[a-zA-Z_][a-zA-Z_0-9]*\}`)
+
 // Parse returns a parsed Lava configuration given an [io.Reader].
 func Parse(r io.Reader) (Config, error) {
+	buf := new(bytes.Buffer)
+	if _, err := buf.ReadFrom(r); err != nil {
+		return Config{}, fmt.Errorf("read config: %w", err)
+	}
+
+	r = strings.NewReader(reEnv.ReplaceAllStringFunc(buf.String(), func(match string) string {
+		return os.Getenv(match[2 : len(match)-1])
+	}))
+
 	dec := yaml.NewDecoder(r)
```

Comment:
This is a style thing more than anything, but you create a `bytes.Buffer` just to read all from `r` and then call `bytes.Buffer.String`, so I think it is more clear to do something like this:

```suggestion
	b, err := io.ReadAll(r)
	if err != nil {
		return Config{}, fmt.Errorf("read config: %w", err)
	}

	s := reEnv.ReplaceAllStringFunc(string(b), func(match string) string {
		return os.Getenv(match[2 : len(match)-1])
	})

	dec := yaml.NewDecoder(strings.NewReader(s))
```

Reply:
thanks, better io.ReadAll